### PR TITLE
Upload unified rendering

### PR DESCRIPTION
**Pull request by TeyaVes**
closes: https://github.com/telerik/kendo-themes-private/issues/48

### Rendering

<details>

<summary> Universal </summary>

```html
<!-- default rendering -->

<div class="k-upload k-upload-async">
  <div class="k-dropzone k-upload-dropzone">
    <div class="k-upload-button-wrap">
      <label for="upload-input"> <!-->optional element<-->
        <button class="k-upload-button k-button k-button-md k-button-solid k-button-solid-base k-rounded-md">
          <span class="k-button-text">Select files...</span>
        </button>
      </label>
      <input id="upload-input">
    </div>
    <span class="k-dropzone-hint k-hidden">Drop files here to upload</span>
    <span class="k-upload-status">
      <span class="k-icon k-i-upload"></span>
      Uploading...
    </span>
  </div>
  <ul class="k-upload-files">
    <!-- single file -->
    <li class="k-file k-file-progress">
      <div class="k-file-single">
        <div class="k-progressbar k-progressbar-horizontal">
          <span class="k-progress-status-wrap k-progress-end"></span>
          <div class="k-selected" style="width: 60%;">
            <span class="k-progress-status-wrap k-progress-end"></span>
          </div>
        </div>
        <span class="k-file-icon-wrapper">
          <span class="k-file-icon k-icon k-i-file-pdf"></span>
          <span class="k-file-state">uploaded</span>
        </span>
        <div class="k-file-info">
          <span class="k-file-name">Document.txt</span>
          <span class="k-file-size">225.68 KB</span>
        </div>
        <div class="k-upload-actions">
          <span class="k-upload-pct">60%</span>
          <button class="k-upload-action k-button k-button-md k-button-flat k-button-flat-base k-rounded-md k-icon-button">
            <span class="k-button-icon k-icon k-i-pause-sm"></span>
          </button>
          <button class="k-upload-action k-button k-button-md k-button-flat k-button-flat-base k-rounded-md k-icon-button">
            <span class="k-button-icon k-icon k-i-cancel"></span>
          </button>
        </div>
      </div>
    </li>
    <!-- multiple files -->
    <li class="k-file k-file-success">
      <div class="k-file-multiple">
        <div class="k-progressbar k-progressbar-horizontal k-hidden">
          <span class="k-progress-status-wrap k-progress-end"></span>
          <div class="k-selected">
            <span class="k-progress-status-wrap k-progress-end"></span>
          </div>
        </div>
        <span class="k-file-icon-wrapper">
          <span class="k-file-icon k-icon k-i-copy"></span>
        </span>
        <div class="k-multiple-files-wrapper">
          <div class="k-file-info">
            <span class="k-file-name">Name.png</span>
            <span class="k-file-size">19.61 KB</span>
          </div>
          ...
          <span class="k-file-summary k-hidden">Total: 4 files, 170.22 KB</span>
          <span class="k-file-validation-message">4 File(s) uploaded successfully</span>
        </div>
        <div class="k-upload-actions">
          <button class="k-upload-action k-button k-button-md k-button-flat k-button-flat-base k-rounded-md k-icon-button">
            <span class="k-button-icon k-icon k-i-close"></span>
          </button>
        </div>
      </div>
    </li>
  </ul>
  <div class="k-actions k-actions-end k-actions-horizontal">
    <button class="k-clear-selected k-button k-button-md k-button-solid k-button-solid-base k-rounded-md">
      <span class="k-button-text">Clear</span>
    </button>
    <button class="k-upload-selected k-button k-button-md k-button-solid k-button-solid-primary k-rounded-md">
      <span class="k-button-text">Update</span>
    </button>
  </div>
</div>
```
</details>

### Changes

1. `k-widget` class removed from `k-upload`
1. all `<em>`,` <strong>`, etc. html elements are replaced with spans or divs
1. `k-upload-dropzone` class added to `k-dropzone`
1. `k-dropzone-hover/hovered `should be replaced with class `k-hover`
1. Upload button and input are no longer nested but are on one level wrapped inside a div (a11y proposal):
    ```
        <div class="k-upload-button-wrap">
          <label for="upload-input"> <!-->optional element<-->
            <button class="k-upload-button k-button k-button-md k-button-solid k-button-solid-base k-rounded-md">
              <span class="k-button-text">Select files...</span>
            </button>
          </label>
          <input id="upload-input">
        </div>
    ```
1. `k-upload-status-total` class removed from `k-upload-status`
1. `k-validation-icon` class is removed from upload-status icon
1. `k-file-error` class added to `k-file-invalid` (see https://github.com/telerik/kendo-themes/pull/4076#issuecomment-1348569208)
1. `k-file-single` and `k-file-multiple` wrapper elements added inside `.k-file` for single and multiple files respectively
1. `k-file-group-wrapper`, `k-file-invalid-group-wrapper`, `k-multiple-files-group-wrapper`and k`-multiple-files-invalid-group-wrapper` are now replaced with `k-file-icon-wrapper`
1. `k-file-group` icon class replaced with `k-file-icon`
1. `k-file-name-size-wrap` class replaced with `k-file-info`
1. `k-file-name-invalid` class removed from `k-file-name`
1. `k-text-success` and `k-text-error `classes removed from `k-validation-message`
1. `k-file-information` class replaced with `k-file-summary`
1. upload actions buttons are no longer wrapped in `k-upload-status` class but in `k-upload-actions` class
1. progress bar replaced with the progressBar component without visible labels

### CODE REVIEW

**Comment by TeyaVes** [comment]
@telerik/kendo-a11y , can you please review the suggested updates on the upload rendering

**Comment by epetrow** [comment]
Just one comment from me, do we need to use `{...htmlAttributes}` in the helpers upload-dropzone, upload-file and upload?

**Review by veselints — APPROVED** [review]

**Comment by TeyaVes** [comment]
> Just two things that I noticed:
> 
> * Do we need to use `{...htmlAttributes}` in the helpers upload-dropzone, upload-file and upload?
> * The color of the close buttons has changed, is that intended?

I've removed the `{...htmlAttributes}` from where it wasn't needed;
The flat buttons changed color as they are no longer in a `k-upload-status` container. This has been synced with the design team.

**Review by epetrow — APPROVED** [review]

File: packages/html/src/upload/upload-file.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import { UploadFileInfo } from './upload-file-info';
+
+export interface UploadFileProps {
+    className?: string;
+    fileStatus?: 'error' | 'progress' | 'success';
+    progress?: number;
+    fileName?: string;
+    fileIcon?: string;
+    fileSize?: string;
+    fileState?: 'uploaded' | 'uploading';
```
Comment (Juveniel) [inline packages/html/src/upload/upload-file.tsx:14]: do we need the file prefix here ? As we are already in the UploadFile context, we can use name, size, etc.

**Review by Juveniel — COMMENTED** [review]
I noticed that in the classic theme, the bottom border of the button is not visible. Could you take a look ?

image
 
 One additional thing - the dropzone text in [the pdf viewer test](tests/_output/bootstrap/pdf-viewer/pdf-viewer-blank-page.png) does not seem centered.

**Comment by TeyaVes** [comment]
> I noticed that in the classic theme, the bottom border of the button is not visible. Could you take a look ?
> 
> image
> 
> One additional thing - the dropzone text in [the pdf viewer test](tests/_output/bootstrap/pdf-viewer/pdf-viewer-blank-page.png) does not seem centered.

The border in the Classic theme is cut out only in the image. I tested it on Mozilla and Chrome and the bottom border is visible in both cases.

File: packages/default/scss/upload/_layout.scss [hunk 2]
```diff
@@ -157,201 +153,62 @@
...
 
-        // old rendering
-        .k-progress {
-            height: $upload-progress-thickness;
-            position: absolute;
-            bottom: 0;
-            left: 0;
-        }
-
         .k-progressbar {
```
Comment (joneff) [inline packages/default/scss/upload/_layout.scss:206]: If you need to, sync with Emo about the progressbar. If needed add a css variable that sets the height of the progressbar.

File: packages/default/scss/upload/_layout.scss [hunk 3]
```diff
@@ -157,201 +153,62 @@
...
-            height: $upload-progress-thickness;
+            height: $kendo-upload-progress-thickness;
             position: absolute;
-            bottom: $upload-item-padding-y / 2;
+            bottom: $kendo-upload-item-padding-y / 2;
             // successed
-            width: calc(100% - #{$upload-item-padding-x} * 2);
+            width: calc(100% - #{$kendo-upload-item-padding-x} * 2);
 
             .k-progress {
```
Comment (joneff) [inline packages/default/scss/upload/_layout.scss:213]: This should probably be removed. We should not customize such styles.

File: packages/default/scss/upload/_layout.scss [hunk 4]
```diff
@@ -361,20 +218,18 @@
...
 
         // RTL styles
         &[dir = "rtl"],
+        [dir = "rtl"] &,
+        &.k-rtl,
         .k-rtl & {
-            .k-dropzone {
+            .k-dropzone,
+            .k-upload-dropzone {
                 text-align: left;
```
Comment (joneff) [inline packages/default/scss/upload/_layout.scss:233]: we can use logical text align.

File: packages/default/scss/upload/_layout.scss [hunk 5]
```diff
@@ -361,20 +218,18 @@
...
 
                 .k-upload-status > .k-icon {
```
Comment (joneff) [inline packages/default/scss/upload/_layout.scss:235]: we can use logical margin

File: packages/default/scss/upload/_theme.scss [hunk 6]
```diff
@@ -2,129 +2,89 @@
...
         .k-file-size,
         .k-file-validation-message,
-        .k-file-information {
+        .k-file-summary {
             color: $subtle-text;
         }
 
-        .k-multiple-files-wrapper .k-file-information {
-            color: $upload-text;
+        .k-multiple-files-wrapper .kk-file-summary {
```
Comment (joneff) [inline packages/default/scss/upload/_theme.scss:86]:
```suggestion
        .k-multiple-files-wrapper .k-file-summary {
```

double kk in selector.

**Review by joneff — CHANGES_REQUESTED** [review]
Few changes to address

**Review by Juveniel — APPROVED** [review]

**Comment by Raisolution** [comment]
> k-file-invalid class is replaced with k-file-error

There is a difference between those two classes. The first one refers to a file that has failed the set restrictions and will not be uploaded at all, while the second one refers to a file which was attempted for upload, but the upload process failed. I am not sure if the two classes attach different styling, but `k-file-invalid` should stay as is, as it has different meaning for the Upload users.

**Comment by TeyaVes** [comment]
> > k-file-invalid class is replaced with k-file-error
> 
> There is a difference between those two classes. The first one refers to a file that has failed the set restrictions and will not be uploaded at all, while the second one refers to a file which was attempted for upload, but the upload process failed. I am not sure if the two classes attach different styling, but `k-file-invalid` should stay as is, as it has different meaning for the Upload users.

We've decided to keep the `k-file-invalid` class along with the `k-file-error` class. The themes will use the `k-file-error` class for styling, but we are keeping the `k-file-invalid` class if the clients need to differentiate between the two

**Review by joneff — APPROVED** [review]

**Pull request merged**